File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        hash_map = {}
        ans = []
        
        def update(k:int) -> None:
            if len(hash_map[k]) == k:
                ans.append(hash_map[k])
                hash_map[k] = []
        
        for i,v in enumerate(groupSizes):
            if v in hash_map:
                hash_map[v].append(i)
            else:
                hash_map[v] = [i]
            update(v)
        
        return ans

File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.go
func groupThePeople(groupSizes []int) [][]int {
    hashMap := make(map[int][]int)
    for i, v := range groupSizes {
        _, ok := hashMap[v]
        if ok {
            hashMap[v] = append(hashMap[v],i)
        } else {
            hashMap[v] = []int {i}
        }
    }
    
    answer := make([][]int,0)
    
    for k1 := range hashMap {
        temp := make([]int,0)
        for _, v2 := range hashMap[k1] {
            temp = append(temp,v2)
            if len(temp) == k1 {
                answer = append(answer, temp)
                temp = []int {}
            }
        }
    }
    
    return answer
}